b2g/chrome/content/remote_control.js

```diff
@@ -31,5 +31,48 @@ const kPrefPresentationDiscoverable = "dom.presentation.discoverable";
 
   Services.prefs.addObserver(kPrefPresentationDiscoverable, setupRemoteControl, false);
 
+  var networkStatusMonitor = {
+    // nsIObserver
+    observe: function(subject, topic, data) {
+      switch (topic) {
+        case "network-active-changed": {
+          if (!subject) {
+            // Stop service when there is no active network
+            remoteControlScope.RemoteControlService.stop();
+            break;
+          }
+
+          // Start service when active network change with new IP address
+          // Other case will be handled by "network:offline-status-changed"
+          if (!Services.io.offline && Services.prefs.getBoolPref(kPrefPresentationDiscoverable)) {
+            importRemoteControlService();
+            remoteControlScope.RemoteControlService.start();
+          }
+          break;
+        }
+        case "network:offline-status-changed": {
+          if (data == "offline") {
+            // Stop service when network status change to offline
+            remoteControlScope.RemoteControlService.stop();
+          } else {
+            if (Services.prefs.getBoolPref(kPrefPresentationDiscoverable)) {
```

Comment:
Hmm....I don't think it will make confuse. if data is offline, it must go in the the first {}, or it will go to the next {}. It looks clear enough for me. Also, I haven't saw this style in gecko yet.